ts/screens/wallet/AddCardScreen.tsx

```diff
@@ -126,29 +126,26 @@ function getCardFromState(state: State): Option<CreditCard> {
 
   const [expirationMonth, expirationYear] = expirationDate.value.split("/");
 
-  if (!CreditCardPan.is(pan.value)) {
+  if (!isPanValid(pan.value)) {
     // invalid pan
     return none;
   }
-  if (
-    !CreditCardExpirationMonth.is(expirationMonth) ||
-    !CreditCardExpirationYear.is(expirationYear)
-  ) {
+  if (!isExpirationDateValid(expirationDate.value)) {
     // invalid date
     return none;
   }
 
-  if (!CreditCardCVC.is(securityCode.value)) {
+  if (!isSecurityCodeValid(securityCode.value)) {
     // invalid cvc
     return none;
   }
 
   const card: CreditCard = {
-    pan: pan.value,
+    pan: pan.value as CreditCardPan,
```

Comment:
ah I see, here's your problem - `CreditCardPan.is` (and all the `.is` functions) are all [type guards](https://basarat.gitbooks.io/typescript/docs/types/typeGuard.html): 

```ts
if(CreditCardPan.is(pan.value)) {
  // pan.value is of type CreditCardPan within this if block
}
```

Since you wrapped the type guards with alias functions, you loose the type information generated by the guards, hence the need to cast. You should instead avoid using the helpers method and instead use the type guards.